domain/modelconfig/service/service.go

```diff
@@ -295,19 +298,22 @@ func (s *Service) UpdateModelConfig(
 // secrets in dqlite
 type dummySecretsBackendProvider struct{}
 
-// dummySpaceProvider implements validators.SpaceProvider and always returns true.
-// TODO (tlm): These needs to be swapped out with an actual checker when we have
-// spaces in dqlite
-type dummySpaceProvider struct{}
-
 // HasSecretsBackend implements validators.SecretBackendProvider
 func (*dummySecretsBackendProvider) HasSecretsBackend(_ string) (bool, error) {
 	return true, nil
 }
 
-// HasSpace implements validators.SpaceProvider
-func (*dummySpaceProvider) HasSpace(_ string) (bool, error) {
-	return true, nil
+// spaceValidator implements validators.SpaceProvider.
+type spaceValidator struct {
+	st State
```

Comment:
I think we should break off a new interface that just describes the concerns of spaceValidator.

Reply:
We already have this interface in the validators, where it's used https://github.com/juju/juju/blob/main/domain/modelconfig/validators/validators.go#L79-L85. Were you thinking about something different?

Reply:
Yeah sorry I mean `State` in this context. At the moment State has the concerns of the service as well. Hope that makes sense. It just a small issue. Not a show stopper.

Reply:
Ohh I see, a new "state" interface only for space checks. On it...

Reply:
@tlm I can have the two "state" interfaces, but this means we'll have to create the service with two states:
```
service.NewService(x,x,state.NewState(...), state.NewState(...))
```
Do you think this is a good change?

Reply:
Oh I think just use composition to get it done?

```
type VState interface {}
type State interface { VState... }
```

Then you can just still take the one arg in `New` It was more about being matter of fact about what the validator needs and not carrying such a big whack of concerns. I more look at it from a testing point of view.

Reply:
done